compiler/rustc_expand/src/mbe/metavar_expr.rs

```diff
@@ -105,11 +112,13 @@ impl MetaVarExpr {
 
 #[derive(Debug, Decodable, Encodable, PartialEq)]
 pub(crate) enum MetaVarExprConcatElem {
-    /// There is NO preceding dollar sign, which means that this identifier should be interpreted
-    /// as a literal.
+    /// Identifier WITHOUT a preceding dollar sign, which means that this identifier should be
+    /// interpreted as a literal.
     Ident(Ident),
-    /// There is a preceding dollar sign, which means that this identifier should be expanded
-    /// and interpreted as a variable.
+    /// For example, a number or a string.
+    Literal(Symbol),
+    /// Identifier WITH a preceding dollar sign, which means that this identifier should be
+    /// expanded and interpreted as a variable.
```

Comment:
So, passing literals through macro variables is not supported, I see.
Is it an intentional restriction or an accident?

Reply:
Errhhhh... Didn't even think this could be a thing.

Let me know if you want this use-case. Otherwise, we can create another unresolved question in the tracking issue.

Reply:
OK. I will do it in a following PR